### PR TITLE
Create separate XML per test file. Repair XML so that reportunit produces correct output.

### PR DESCRIPTION
**Pull request by m-kostrzewa**
Depends on https://github.com/Juniper/contrail-windows-ci/pull/32

### CODE REVIEW

File: CIScripts/NUnitReportFixup/Repair-NUnitReport.Tests.ps1 [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+
+            $ActualOutput = Repair-NUnitReport -InputData $TestData
+            $ActualOutput | Should BeExactly $ExpectedOutput
+        }
+
+    }
+
+    Context "test name cleanup" {
+        It "makes names of tests more readable" {
+
```
Comment (m-kostrzewa) [inline CIScripts/NUnitReportFixup/Repair-NUnitReport.Tests.ps1]: todo in another PR

**Comment by sodar** [comment]
recheck no bug

**Comment by krdln** [comment]
* [Works when there are no test reports](http://10.84.12.26:8080/job/WinContrail/job/contrail-win-ci-gh/981/console)

File: CIScripts/GenerateTestReport.ps1 [hunk 2]
```diff
@@ -1,13 +1,42 @@
...
-        [Parameter(Mandatory = $true)] [String[]] $XmlReports
+        [Parameter(Mandatory = $true)] [String] $XmlReportDir,
+        [Parameter(Mandatory = $true)] [string] $OutputDir
     )
 
-    $XmlReports | ForEach-Object {
-        if (Test-Path $_) {
-            ReportUnit.exe $_
-        }
+    $RawDir = "$OutputDir/raw_NUnit"
```
Comment (sodar) [inline CIScripts/GenerateTestReport.ps1]: Maybe `FixedReportsDir` is a better than `RawDir`.

File: CIScripts/GenerateTestReport.ps1 [hunk 3]
```diff
@@ -1,13 +1,42 @@
...
     }
+    ReportUnit.exe $RawDir
+
+    Move-Item "$RawDir/*.html" $PrettyDir
+
+    $Xmls = Get-ChildItem $RawDir | Foreach-Object { $_.FullName.split('\')[-2 .. -1] -join "/" }
+    @{
+        xml_reports = $Xmls
+        html_report = "$PrettyDirName/Index.html"
+    } | ConvertTo-Json -Depth 10 | Out-File "$OutputDir/reports-locations.json" -Encoding "utf8"
```
Comment (sodar) [inline CIScripts/GenerateTestReport.ps1]: We think that generating JSON could be extracted to separate function.

File: CIScripts/GenerateTestReport.ps1 [hunk 4]
```diff
@@ -1,13 +1,42 @@
...
+
+    New-Item -Type Directory -Force $RawDir | Out-Null
+    New-Item -Type Directory -Force $PrettyDir | Out-Null
+
+    foreach ($ReportFile in Get-ChildItem $XmlReportDir -Filter *.xml) {
+        [string] $Content = Get-Content $ReportFile.FullName
+        $FixedContent = Repair-NUnitReport -InputData $Content
+        $FixedContent | Out-File "$RawDir/$($ReportFile.Name)" -Encoding "utf8"
     }
+    ReportUnit.exe $RawDir
```
Comment (sodar) [inline CIScripts/GenerateTestReport.ps1]: We think that fixing XMLs could be extracted to separate function.

File: CIScripts/NUnitReportFixup/Repair-NUnitReport.ps1 [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+        $_.ParentNode.ParentNode
+    }
+    return ,$Arr
+}
+
+function Move-NodesToRootResultsNode {
+    Param([Parameter(Mandatory = $true)] [AllowEmptyCollection()]
+          [System.Xml.XmlElement[]] $Nodes)
+    $Nodes | ForEach-Object {
+        $XML.'test-results'.AppendChild($_)
```
Comment (sodar) [inline CIScripts/NUnitReportFixup/Repair-NUnitReport.ps1]: We think `XML` variable should be passed as parameter explicitly.

**Comment by sodar** [comment]
DS + JI

**Pull request merged**